### ISSUE TITLE
This crate is completely unsafe

### ISSUE THREAD

**Issue by jasondyoungberg** [LDeakin/unsafe_cell_slice#1]
The mere existence of two mutable references pointing to same object is undefined behavior. The example code, even without the writes, is undefined behavior. There is no possible way for this crate to ever be safe.

**Comment by LDeakin** [comment]
Thanks for raising this @jasondyoungberg. I am curious about your thoughts on #2. Does this address the unsoundness in your opinion? Miri does not find any faults with this.

```rust
let mut data = vec![0u8; 2];
{
    let data = UnsafeCellSlice::new(&mut data);
    let data_a: &mut [u8] = unsafe { data.index_mut(0..1) };
    let data_b: &mut [u8] = unsafe { data.index_mut(1..2) };
    data_a[0] = 0;
    data_b[0] = 1;
}
assert_eq!(data[0], 0);
assert_eq!(data[1], 1);
```

**Comment by jasondyoungberg** [comment]
That should be the correct way to do this. The user simply needs to make sure they never create overlapping references

**Comment by jasondyoungberg** [comment]
Maybe I was a little extreme with my first message... sorry about that

**Comment by LDeakin** [comment]
Closed with #2.
